crates/sequencer/src/serde.rs

```diff
@@ -0,0 +1,101 @@
+use blockifier::{
+    execution::contract_class::ContractClass, state::cached_state::ContractStorageKey,
+};
+use rustc_hash::FxHashMap;
+use serde::{Deserialize, Serialize};
+use starknet_api::{
+    core::{ClassHash, CompiledClassHash, ContractAddress, Nonce},
+    hash::StarkFelt,
+};
+
+use crate::state::State;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum SerializationError {
+    #[error("{reason:?}")]
+    IoError {
+        reason: String,
+        context: std::io::Error,
+    },
+    #[error("{reason:?}")]
+    SerdeJsonError {
+        reason: String,
+        context: serde_json::Error,
+    },
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+pub struct SerializableState {
+    pub classes: FxHashMap<ClassHash, ContractClass>,
+    pub compiled_classes_hash: FxHashMap<ClassHash, CompiledClassHash>,
+    pub contracts: FxHashMap<ContractAddress, ClassHash>,
+    #[serde(with = "serialize_contract_storage")]
+    pub storage: FxHashMap<ContractStorageKey, StarkFelt>,
+    pub nonces: FxHashMap<ContractAddress, Nonce>,
+}
+impl From<&State> for SerializableState {
+    fn from(state: &State) -> Self {
+        Self {
+            classes: state.classes.clone(),
+            compiled_classes_hash: state.compiled_class_hashes.clone(),
+            contracts: state.contracts.clone(),
+            storage: state.storage.clone(),
+            nonces: state.nonces.clone(),
+        }
+    }
+}
```

Comment:
Makes sense, I did this to avoid cloning, but now I realize that at the end of the day we seem to be cloning the fields anyway!

Lemme change the implementation

Reply:
resolved via [this](https://github.com/kkrt-labs/ef-tests/pull/555/commits/4d5fb42d2758e4aa15fd339359ea19e4e27a0039) commit.